eth-client/src/lib.rs

```diff
@@ -0,0 +1,243 @@
+mod error;
+use error::VerifierError;
+use ethcontract::prelude::*;
+use primitive_types::U256;
+use std::env;
+ethcontract::contract!("solidity_verifier/hardhat/artifacts/contracts/vs_change/validator_change_verifier.sol/ValidatorSetChangeVerifier.json");
+ethcontract::contract!("solidity_verifier/hardhat/artifacts/contracts/message_sent/message_sent_verifier.sol/MessageSentVerifier.json");
+
+pub struct ContractVerifiers {
+    vs_change_vrf: validator_set_change_verifier::Contract,
+    msg_sent_vrf: message_sent_verifier::Contract,
+}
+
+impl ContractVerifiers {
+    pub fn new(
+        url: &str,
+        address_vs_change: &str,
+        address_msg_sent: &str,
+    ) -> Result<Self, VerifierError> {
+        let http = Http::new(url).map_err(|_| VerifierError::ErrorInHTTPTransport)?;
+        let web3 = Web3::new(http);
+        let address_vs_change = address_vs_change
+            .parse::<web3::types::Address>()
+            .map_err(|_| VerifierError::WrongAddress)?;
+
+        let address_msg_sent = address_msg_sent
+            .parse::<web3::types::Address>()
+            .map_err(|_| VerifierError::WrongAddress)?;
+
+        Ok(ContractVerifiers {
+            vs_change_vrf: ValidatorSetChangeVerifier::at(&web3, address_vs_change),
+            msg_sent_vrf: MessageSentVerifier::at(&web3, address_msg_sent),
+        })
+    }
+
+    pub async fn verify_vs_change(
+        &self,
+        p_a: [U256; 2],
+        p_b: [[U256; 2]; 2],
+        p_c: [U256; 2],
+        validator_set: [U256; 5],
+        nonce_id: U256,
+    ) -> Result<bool, VerifierError> {
+        let account = {
+            let pk = env::var("PK").expect("PK is not set");
```

Comment:
I think it's better to turn it into the parameter to get it from args/env in relayer